Partition stays in force configuration after force failover

In some failed e2e multi-region failover test, failback took long time because a partition did not come out of forced configuration for almost 1hr. Eventually it succeeds though, but the test failed because of the timeout. It is not expected to take so long. Coming out of force configuration should be done with in seconds after force configuration is succeeded.
 
Example of such a run: https://camunda.slack.com/archives/C013MEVQ4M9/p1711523599459309 

We can see the following error repeated for almost 1 hour. Then eventually the operation succeeds (probably after a leader change.)
```
java.util.concurrent.CompletionException: io.atomix.primitive.PrimitiveException$Unavailable: Force configuration change is in progress. Cannot accept request from 3 which is not a member of the new configuration.
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniRun.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(Unknown Source) ~[?:?]
	at io.atomix.raft.impl.ReconfigurationHelper.lambda$joinWithRetry$5(ReconfigurationHelper.java:135) ~[zeebe-atomix-cluster-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$UniWhenComplete.tryFire(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.CompletableFuture$Completion.run(Unknown Source) ~[?:?]
	at io.atomix.utils.concurrent.SingleThreadContext$WrappedRunnable.run(SingleThreadContext.java:178) ~[zeebe-atomix-utils-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	at java.base/java.lang.Thread.run(Unknown Source) [?:?]
Caused by: io.atomix.primitive.PrimitiveException$Unavailable: Force configuration change is in progress. Cannot accept request from 3 which is not a member of the new configuration.
	at io.atomix.raft.RaftError$Type$5.createException(RaftError.java:139) ~[zeebe-atomix-cluster-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.atomix.raft.RaftError.createException(RaftError.java:65) ~[zeebe-atomix-cluster-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	at io.atomix.raft.impl.ReconfigurationHelper.lambda$joinWithRetry$5(ReconfigurationHelper.java:133) ~[zeebe-atomix-cluster-8.5.0-SNAPSHOT.jar:8.5.0-SNAPSHOT]
	... 10 more
```


related to #16126

--- a/zeebe/atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java
+++ b/zeebe/atomix/cluster/src/main/java/io/atomix/raft/roles/LeaderRole.java
@@ -230,19 +230,6 @@ public final class LeaderRole extends ActiveRole implements ZeebeLogAppender {
   @Override
   public CompletableFuture<ForceConfigureResponse> onForceConfigure(
       final ForceConfigureRequest request) {
-    final Configuration currentConfiguration = raft.getCluster().getConfiguration();
-    if (currentConfiguration != null // this is not expected in a leader
-        && request.newMembers().equals(currentConfiguration.allMembers())) {
-      // It is likely that the previous force configure was successfully completed, and this is a
-      // retry. So just respond success.
-      return CompletableFuture.completedFuture(
-          logResponse(
-              ForceConfigureResponse.builder()
-                  .withStatus(Status.OK)
-                  .withIndex(currentConfiguration.index())
-                  .withTerm(currentConfiguration.term())
-                  .build()));
-    }
 
     // Do not force-configure when you are leader.
     raft.transition(Role.FOLLOWER);

--- a/zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java
+++ b/zeebe/atomix/cluster/src/test/java/io/atomix/raft/ReconfigurationTest.java
@@ -80,6 +80,10 @@ final class ReconfigurationTest {
         .findAny();
   }
 
+  private static Optional<RaftServer> getFollower(final RaftServer... servers) {
+    return Arrays.stream(servers).filter(RaftServer::isFollower).findAny();
+  }
+
   private static AppendResult appendEntry(final LeaderRole leader) {
     final var result = new AppendResult();
     leader.appendEntry(-1, -1, ByteBuffer.wrap(new byte[0]), result);
@@ -738,6 +742,68 @@ final class ReconfigurationTest {
       // then
       awaitLeader(m1, m2);
       assertThat(m1.getContext().isLeader()).isTrue();
+    }
+
+    @Test
+    void forceReconfigureIsIdempotentWhenRetriedViaAFollower() {
+      // given
+      m2.forceConfigure(newMembers()).join();
+      m3.shutdown().join();
+      m4.shutdown().join();
+      awaitLeader(m1, m2);
+      Awaitility.await("Both members have come out of force configuration")
+          .untilAsserted(
+              () -> {
+                assertThat(m2.getContext().getCluster().getConfiguration().force()).isFalse();
+                assertThat(m1.getContext().getCluster().getConfiguration().force()).isFalse();
+              });
+
+      // when
+      final var follower = getFollower(m1, m2).orElseThrow();
+      follower.forceConfigure(newMembers()).join();
+
+      // then
+      Awaitility.await("Both members have come out of force configuration")
+          .untilAsserted(
+              () -> {
+                assertThat(m2.getContext().getCluster().getConfiguration().force())
+                    .describedAs("Member 2 has come out of force configuration")
+                    .isFalse();
+                assertThat(m1.getContext().getCluster().getConfiguration().force())
+                    .describedAs("Member 1 has come out of force configuration")
+                    .isFalse();
+              });
+    }
+
+    @Test
+    void forceReconfigureIsIdempotentWhenRetriedViaLeader() {
+      // given
+      m2.forceConfigure(newMembers()).join();
+      m3.shutdown().join();
+      m4.shutdown().join();
+      awaitLeader(m1, m2);
+      Awaitility.await("Both members have come out of force configuration")
+          .untilAsserted(
+              () -> {
+                assertThat(m2.getContext().getCluster().getConfiguration().force()).isFalse();
+                assertThat(m1.getContext().getCluster().getConfiguration().force()).isFalse();
+              });
+
+      // when
+      final var leader = Stream.of(m1, m2).filter(RaftServer::isLeader).findAny().orElseThrow();
+      leader.forceConfigure(newMembers()).join();
+
+      // then
+      Awaitility.await("Both members have come out of force configuration")
+          .untilAsserted(
+              () -> {
+                assertThat(m2.getContext().getCluster().getConfiguration().force())
+                    .describedAs("Member 2 has come out of force configuration")
+                    .isFalse();
+                assertThat(m1.getContext().getCluster().getConfiguration().force())
+                    .describedAs("Member 1 has come out of force configuration")
+                    .isFalse();
+              });
     }
 
     private Map<MemberId, Type> newMembers() {

